File: src/ios/UMengStatistics.m
/********* UMengStatistics.m Cordova Plugin Implementation *******/

#import <Cordova/CDV.h>

#import "MobClick.h"

static BOOL isInit = false;

@interface UMengStatistics : CDVPlugin {
  // Member variables go here.
}

- (void)pageStart:(CDVInvokedUrlCommand*)command;
- (void)pageEnd:(CDVInvokedUrlCommand*)command;
- (void)pageEvent:(CDVInvokedUrlCommand*)command;
@end

@implementation UMengStatistics

- (void)initMobClick{
    if(!isInit){
        isInit = true;
        CDVViewController* viewController = (CDVViewController*)self.viewController;
        NSString* umengAppKey = [viewController.settings objectForKey:@"umengappkey"];
        [MobClick startWithAppkey:umengAppKey reportPolicy:SEND_INTERVAL channelId:@"appstore"];
    }
}

- (void)pageStart:(CDVInvokedUrlCommand*)command
{
    [self initMobClick];
    NSString* pageName = [command.arguments objectAtIndex:0];
    [MobClick beginLogPageView:pageName];
}

- (void)pageEnd:(CDVInvokedUrlCommand*)command
{
    [self initMobClick];
    NSString* pageName = [command.arguments objectAtIndex:0];
    [MobClick endLogPageView:pageName];
}

-(void)pageEvent:(CDVInvokedUrlCommand*)command
{
    [self initMobClick];
    NSString* eventId = [command.arguments objectAtIndex:0];
    [MobClick event:eventId];
}

@end

File: www/UMengStatistics.js
var exec = require('cordova/exec');

var noop = function(){};
exports.exec = function(action, arg0, success, error) {
    exec(success, error, "UMengStatistics", action, [arg0]);
};

exports.pageStart = function(pageName){
	exports.exec('pageStart', pageName, noop, noop);
};

exports.pageEnd = function(pageName){
	exports.exec('pageEnd', pageName, noop, noop);
};

exports.pageEvent = function(eventId){
	exports.exec('pageEvent', eventId, noop, noop);
};